cmd/scollector/collectors/vsphere.go

```diff
@@ -181,9 +183,10 @@ func vsphereHost(v *vsphere.Vsphere, md *opentsdb.MultiDataPoint) error {
 							return err
 						}
 						if t.IdentiferType.Key == "ServiceTag" {
-							metadata.AddMeta("", tags, "serialNumber", t.IdentiferValue, false)
+							lastServiceTag = t.IdentiferValue
 						}
 					}
+					metadata.AddMeta("", tags, "serialNumber", lastServiceTag, false)
```

Comment:
lastServiceTag could be empty if the for loop never ran or something. Instead, we should just have  `serviceTagDone := false` before the for loop, then if the bool is false, set the metadata where it is now, then set the bool to true.